src/plugins/navigation_embeddable/public/components/navigation_embeddable_panel_editor.tsx

```diff
@@ -227,11 +214,70 @@ const NavigationEmbeddablePanelEditor = ({
           <EuiFlexItem grow={false}>
-            <EuiButtonEmpty onClick={onClose} iconType="cross">
+            <EuiButtonEmpty onClick={onClose} iconType="cross" flush="left">
               {NavEmbeddableStrings.editor.getCancelButtonLabel()}
             </EuiButtonEmpty>
           </EuiFlexItem>
-          <EuiFlexItem grow={false}>{saveButtonComponent}</EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiFlexGroup>
+              {!isByReference ? (
+                <EuiFlexItem grow={false} css={{ 'margin-left': 'auto' }}>
+                  <EuiToolTip
+                    repositionOnScroll={false}
+                    position="top"
+                    content={
+                      <p>
+                        {NavEmbeddableStrings.editor.panelEditor.getAddToDashboardButtonTooltip()}
+                      </p>
+                    }
+                  >
+                    <EuiButton
+                      disabled={orderedLinks.length === 0}
+                      isLoading={isSaving}
+                      onClick={async () => {
+                        setIsSaving(true);
+                        await onAddToDashboard(orderedLinks);
+                        setIsSaving(false);
+                      }}
+                    >
+                      {NavEmbeddableStrings.editor.panelEditor.getAddToDashboardButtonLabel()}
+                    </EuiButton>
+                  </EuiToolTip>
+                </EuiFlexItem>
+              ) : null}
+              {!initialLinks || isByReference ? (
+                <EuiFlexItem grow={false}>
+                  <EuiToolTip
+                    repositionOnScroll={false}
+                    position="top"
+                    content={
+                      <p>
+                        {initialLinks
+                          ? NavEmbeddableStrings.editor.panelEditor.getUpdateLibraryItemButtonTooltip()
+                          : NavEmbeddableStrings.editor.panelEditor.getSaveToLibraryButtonTooltip()}
+                      </p>
+                    }
+                  >
+                    <EuiButton
+                      fill
+                      iconType="folderCheck"
+                      disabled={orderedLinks.length === 0}
+                      isLoading={isSaving}
+                      onClick={async () => {
+                        setIsSaving(true);
+                        await onSaveToLibrary(orderedLinks);
+                        setIsSaving(false);
+                      }}
+                    >
+                      {initialLinks
+                        ? NavEmbeddableStrings.editor.panelEditor.getUpdateLibraryItemButtonLabel()
+                        : NavEmbeddableStrings.editor.panelEditor.getSaveToLibraryButtonLabel()}
+                    </EuiButton>
+                  </EuiToolTip>
+                </EuiFlexItem>
+              ) : null}
+            </EuiFlexGroup>
+          </EuiFlexItem>
```

Comment:
We could make this `EuiFlyoutFooter` into separate components. But I think it is out-of-scope for this PR until we have user testing to answer the question "Is it better to use two buttons in the footer for by-value / by-value creation, or do we want to use the originally proposed `EuiSwitch`"?